### PR TITLE
Web app

### PR DESCRIPTION
**Pull request by YevheniiLohatskyi**
yet not finished, will finish tomorrow in the morning

### CODE REVIEW

File: src/main/java/com/homework/webapp/exception/NotFoundEntityException.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package com.homework.webapp.exception;
+
+import java.util.UUID;
+
+public class NotFoundEntityException extends Exception {
+    public NotFoundEntityException(){
+    super("No user found");
```
Comment (Anderwerty) [inline src/main/java/com/homework/webapp/exception/NotFoundEntityException.java]: Why do you use such a message for NotFoundEntityException?

File: src/main/java/com/homework/webapp/controller/ManufacturerController.java [hunk 2]
```diff
@@ -0,0 +1,4 @@
+package com.homework.webapp.controller;
+
+public class ManufacturerController {
```
[the remark this answers is not in the document]
Comment (YevheniiLohatskyi) [reply src/main/java/com/homework/webapp/controller/ManufacturerController.java]: yes, I just haven't written it yet

File: src/main/java/com/homework/webapp/mapper/ProductMapper.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    public List<Product> mapProductEntitiesToProducts(List<ProductEntity> entities){
+        return entities.stream()
+                .map(this::mapProductEntityToProduct)
+                .collect(toList());
+    }
+
+    public ProductEntity mapProductToProductEntity(Product product){
+        ProductEntity entity = new ProductEntity()
```
[the remark this answers is not in the document]
Comment (YevheniiLohatskyi) [reply src/main/java/com/homework/webapp/mapper/ProductMapper.java]: It was like that in the beginning, then I tried to add class BaseEntity which has a field UUID id, so I changed it to how it looks now. And forgot to revert these changes when I deleted DaseEntity class

File: src/main/java/com/homework/webapp/exception/NotFoundEntityException.java (hunk 1 above)
Comment (YevheniiLohatskyi) [reply src/main/java/com/homework/webapp/exception/NotFoundEntityException.java]: Copy-pasted it from previous homework and forgot to change it to a more general message

File: pom.xml [hunk 4]
```diff
@@ -8,5 +8,114 @@
...
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-test</artifactId>
+            <version>2.1.4.RELEASE</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-test</artifactId>
+            <version>4.3.7.RELEASE</version>
```
Comment (Anderwerty) [inline pom.xml]: Could we use version 5.1.6RELEASE here?

File: pom.xml [hunk 5]
```diff
@@ -8,5 +8,114 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-autoconfigure</artifactId>
+            <version>2.1.4.RELEASE</version>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.security</groupId>
+            <artifactId>spring-security-test</artifactId>
+            <version>5.1.5.RELEASE</version>
```
Comment (Anderwerty) [inline pom.xml]: Could we use version 5.1.6RELEASE here?

File: pom.xml [hunk 6]
```diff
@@ -8,5 +8,114 @@
...
+        </dependency>
+        <dependency>
+            <groupId>mysql</groupId>
+            <artifactId>mysql-connector-java</artifactId>
+            <version>8.0.15</version>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-autoconfigure</artifactId>
+            <version>2.1.4.RELEASE</version>
```
Comment (Anderwerty) [inline pom.xml]: 2.1.6.RELEASE?

File: pom.xml [hunk 7]
```diff
@@ -8,5 +8,114 @@
...
+        <dependency>
+            <groupId>org.springframework.security</groupId>
+            <artifactId>spring-security-test</artifactId>
+            <version>5.1.5.RELEASE</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-security</artifactId>
+            <version>2.1.4.RELEASE</version>
```
Comment (Anderwerty) [inline pom.xml]: 2.1.6.RELEASE?

File: pom.xml [hunk 8]
```diff
@@ -8,5 +8,114 @@
...
+        </dependency>
+        <dependency>
+            <groupId>javax.validation</groupId>
+            <artifactId>validation-api</artifactId>
+            <version>2.0.1.Final</version>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-web</artifactId>
+            <version>2.1.4.RELEASE</version>
```
Comment (Anderwerty) [inline pom.xml]: 2.1.6.RELEASE?

File: pom.xml [hunk 9]
```diff
@@ -8,5 +8,114 @@
...
+            <scope>compile</scope>
+        </dependency>
+        <dependency>
+            <groupId>javax.servlet</groupId>
+            <artifactId>jstl</artifactId>
+            <version>1.2</version>
+        </dependency>
+    </dependencies>
 </project>
```
[the remark this answers is not in the document]
Comment (Anderwerty) [reply pom.xml]: Add a blank line at the end of the file.

File: src/main/java/com/homework/webapp/controller/ManufacturerController.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Anderwerty) [reply src/main/java/com/homework/webapp/controller/ManufacturerController.java]: Did you finish it?

File: src/main/java/com/homework/webapp/mapper/ProductMapper.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (Anderwerty) [reply src/main/java/com/homework/webapp/mapper/ProductMapper.java]: Try to change it.

File: src/main/java/com/homework/webapp/configuration/WebSecurityConfig.java [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+
+@Configuration
+@EnableWebSecurity
+public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
+   @Autowired
+    private UserDetailsService userDetailsService;
```
Comment (Anderwerty) [inline src/main/java/com/homework/webapp/configuration/WebSecurityConfig.java]: Why not use the constructor for dependency injection?

File: src/main/java/com/homework/webapp/dto/Role.java [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+
+import java.util.UUID;
+
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+@Accessors(chain = true)
+public class Role {
+    private Long id;
```
Comment (Anderwerty) [inline src/main/java/com/homework/webapp/dto/Role.java]: Why did you need both id and uuid?
